Some tweaks

Izak8 opened the pull request:
Fix compile issue related to `c11` standard instead of `gnu89`. Changes the C standard to be `gnu89` since driving code in `test-mergesort.c` references a function `random()` which must be a GNU extension. Also adds the binaries `test-mergesort` and all the unit tests `*.test` to `.gitignore`

Izak8 committed:
chore: add additional paths to `.gitignore`

Includes `*.test` and `test-mergesort` to ignore the binaries produced
by the makefile.

Izak8 committed:
build: it turns out gnu89 _was_ necessary

The driving code in `test-mergesort.c` references a function `random()`
which is evidently a GNU extension. Ugh.

Matthew-ew reviewed (dismissed):
she gnu on my 89 til i c11

Izak8 committed:
test: regoranise test files

All test sources have been moved to the `tests` subdirectory.
Tests are compiled by issuing the `make tests` command (the
recipe name changed).

Matthew-ew reviewed (approved):
he tweakin bruh
